torchtune/datasets/_finetune.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Callable, Dict, List, Mapping, Optional
+
+import numpy as np
+
+from datasets import load_dataset
+from torch.utils.data import Dataset
+from torchtune.config._utils import _get_component_from_path
+from torchtune.data import CROSS_ENTROPY_IGNORE_IDX, Message
+from torchtune.modules.transforms import Transform
+
+
+class FinetuneDataset(Dataset):
```

Comment:
Not a huge fan of this name - why not something like `CustomMultiturnDataset`?

Reply:
Ah I see this is used for `Cauldron`, which I don't think I would count as a "custom" multiturn dataset so maybe the name for this should be `MultiturnChatDataset`??